In [3]:
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeRegressor
from sklearn.metrics import mean_absolute_error
from sklearn.model_selection import train_test_split
# save filepath to variable for easier access
melbourne_file_path = './input/melb_data.csv'
# read the data and store data in DataFrame titled melbourne_data
melbourne_data = pd.read_csv(melbourne_file_path) 



In [4]:
# Filter rows with missing price values
filtered_melbourne_data = melbourne_data.dropna(axis=0)

Data Split
The scikit-learn library has a function train_test_split to break up the data into two pieces. We'll use some of that data as training data to fit the model, and we'll use the other data as validation data to calculate mean_absolute_error.

In [6]:

# Choose target and features
y = filtered_melbourne_data.Price
melbourne_features = ['Rooms', 'Bathroom', 'Landsize', 'BuildingArea', 
                        'YearBuilt', 'Lattitude', 'Longtitude']
X = filtered_melbourne_data[melbourne_features]
from sklearn.model_selection import train_test_split

# split data into training and validation data, for both features and target
# The split is based on a random number generator. Supplying a numeric value to
# the random_state argument guarantees we get the same split every time we
# run this script.
train_X, val_X, train_y, val_y = train_test_split(X, y, random_state = 0)


We build a random forest model similarly to how we built a decision tree in scikit-learn - this time using the RandomForestRegressor class instead of DecisionTreeRegressor.

There are many metrics for summarizing model quality, but we'll start with one called Mean Absolute Error (also called MAE). Let's break down this metric starting with the last word, error.
To calculate MAE, we use mean_absolute_error 

In [13]:
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error

forest_model = RandomForestRegressor(random_state=1)
forest_model.fit(train_X, train_y)
melb_preds = forest_model.predict(val_X)
print(mean_absolute_error(val_y, melb_preds))

191669.7536453626


In [24]:
print("Making predictions for the following 5 houses:")
print(X.head())
print("The predictions are")
print(forest_model.predict(val_X.head()))
print("The target values are")
melbourne_features_price = filtered_melbourne_data[['Rooms', 'Bathroom', 'Landsize', 'BuildingArea', 'YearBuilt', 'Lattitude', 'Longtitude', 'Price']]
print(melbourne_features_price.head())

Making predictions for the following 5 houses:
   Rooms  Bathroom  Landsize  BuildingArea  YearBuilt  Lattitude  Longtitude
1      2       1.0     156.0          79.0     1900.0   -37.8079    144.9934
2      3       2.0     134.0         150.0     1900.0   -37.8093    144.9944
4      4       1.0     120.0         142.0     2014.0   -37.8072    144.9941
6      3       2.0     245.0         210.0     1910.0   -37.8024    144.9993
7      2       1.0     256.0         107.0     1890.0   -37.8060    144.9954
The predictions are
[ 947155.  540290.  976405. 1492295.  662815.]
The target values are
   Rooms  Bathroom  Landsize  BuildingArea  YearBuilt  Lattitude  Longtitude  \
1      2       1.0     156.0          79.0     1900.0   -37.8079    144.9934   
2      3       2.0     134.0         150.0     1900.0   -37.8093    144.9944   
4      4       1.0     120.0         142.0     2014.0   -37.8072    144.9941   
6      3       2.0     245.0         210.0     1910.0   -37.8024    144.9993   
7 

Plot the mean absolute error (MAE) of the validation dataset versus the training dataset

mean_absolute_error is still high (191669.7536453626). As we can see the prediction that our model gives is not very accurate 